enhancements/node-artifacts/certificates-and-pull-secrets.md

```diff
@@ -0,0 +1,199 @@
+  `/node/var/lib/kubelet/config.json` for pull secrets and 
+  `/node/etc/docker/certs.d` for the docker certificates. Something similar to:
+
+```
+volumeMounts:
+- mountPath: /node/etc/docker/certs.d
+  name: node-docker-certificates
+- mountPath: /node/etc/pki/ca-trust/extracted/pem
+  name: node-certificates
+- mountPath: /node/var/lib/kubelet/config.json
+  name: node-pullsecrets
+volumes:
+- hostPath:
+    path: /etc/docker/certs.d
+    type: Directory
+  name: node-docker-certificates
+- hostPath:
+    path: /var/lib/kubelet/config.json
+    type: File
+  name: node-pullsecrets
+- hostPath:
+    path: /etc/pki/ca-trust/extracted/pem
+    type: Directory
+```
+
+- Copy node's certificates to `anchor` destination inside `openshift-apiserver`
+  pod before running the binary(`/etc/pki/ca-trust/source/anchors/`).
+- Run `update-ca-trust` and just then start the binary.
+- When user requires to import an image stream, `openshift-apiserver` would
+  merge all pull secrets found in `/node/var/lib/kubelet/config.json` with
+  other pull secrets that may exist on the namespace.
+- Hook a `watcher` to observe certificates changes on node's mounted directories,
+  running `update-ca-trust` when something changes.
+
+#### Builds
+
+- `controller-manager` would mount these three locations inside `build` pod 
+  as `hostPath`(similar to what is done for Image Stream Import).
+- Run `update-ca-trust` before starting the build.
+- Builder image parses node's pull secrets and uses them during build,
+  merging with other pull secrets that may exist on the same namespace.
+
+### Risks and Mitigations
+
+#### Pull secrets inside build pod may be visible to the user
+
+Cluster wise secrets mounted inside the builder pod may be a security risk
+as the user may be able to shell into the pod and copy them. Another possible
+risk is that the user may copy these secrets into a resulting image.
+
+Mitigations:
+
+As far as I verified(and this needs to be once more tested) it is impossible
+to spawn a shell inside the builder pods. I also tried to copy the secrets from
+builder's filesystem into a resulting image but it failed.
+
+#### Multiple certificate sources overwrite or redundancy
+
+As today we mount some certificates inside `openshift-apiserver` and `builder`
+pods, this process needs to be reviewed as some parts may be redundant.
+
+Mitigations:
+
+Go through all mounted certificates on `openshift-apiserver` and `builder`
+pods and make sure that we still need all of them as some of the certificates
+may be already part of node's certificates now. Node's certificates will be the
+starting point. Any other certificate may still be copied inside the pod but we
+will have to change the process to use `update-ca-trust` binary to merge the
+certificates instead of just copying them.
+
+
+## Design Details
+
+### Test Plan
+
+- Attempt to import images streams from `registry.redhat.io` without provide
+  any other pull secret.
```

Comment:
need to do this in a new/empty project.  (if you try it in the openshift project, it'll work today even without this enhancement because we put the pull-secret in that project).

Reply:
Yes, fixing it.